pkg/server/skydb/recordutil/recordutil.go

```diff
@@ -328,21 +328,33 @@ func RecordSaveHandler(req *RecordModifyRequest, resp *RecordModifyResponse) sky
 	makeAssetsCompleteAndInjectSigner(db, req.Conn, records, req.AssetStore)
 
 	// execute after save hooks
-	if req.HookRegistry != nil {
-		records = executeRecordFunc(records, resp.ErrMap, func(record *skydb.Record) (err skyerr.Error) {
-			originalRecord, _ := originalRecordMap[record.ID]
-			err = req.HookRegistry.ExecuteHooks(req.Context, hook.AfterSave, record, originalRecord)
-			if err != nil {
+	records = hookTriggerer.triggerIfRegistered(records, hook.AfterSave, true)
+
+	resp.SavedRecords = records
+	resp.SchemaUpdated = schemaExtended
+
+	return nil
+}
+
+type saveHookTriggerer struct {
+	Req               *RecordModifyRequest
+	Resp              *RecordModifyResponse
+	OriginalRecordMap map[skydb.RecordID]*skydb.Record
+}
+
+func (t saveHookTriggerer) triggerIfRegistered(records []*skydb.Record, kind hook.Kind, shouldLogErr bool) []*skydb.Record {
```

Comment:
> This method may or may not trigger the hook, and I think this should be reflected in the method name.

No. All code does or does not do something based on condition. The checking being included in the function doesn’t justify the name to include this fact.